File: 1710-maximum-units-on-a-truck/1710-maximum-units-on-a-truck.py
class Solution:
    def maximumUnits(self, boxTypes: List[List[int]], truckSize: int) -> int:
        boxTypes.sort( key = lambda x: (-x[1],x[0]))
        
        bxload = uload = 0 
        for bx , un in boxTypes: 
            if bxload < truckSize: 
                if bxload +bx < truckSize: 
                    bxload += bx 
                    uload += bx*un 
                else: 
                    dif = truckSize - bxload 
                    uload += un*dif 
                    break 
    
            else:
                break 
        
        return uload 

File: 1710-maximum-units-on-a-truck/1710-maximum-units-on-a-truck.java
class Solution {
    public int maximumUnits(int[][] boxTypes, int truckSize) {
        
        

        int[] buk = new int[1001] ; 
        for(int[] bx: boxTypes){
            buk[bx[1]] += bx[0] ; } 

        int bxload = 0; 
        for(int ut =1000 ; ut > 0 ; ut--){
            if( buk[ut] > 0 ) { 
                int dif = Math.min( buk[ut] , truckSize ) ; 
                bxload += ut*dif ; 
                truckSize -= dif ; 
                if(truckSize <=0){ 
                    return bxload ; 
                }
            }
        }
        return bxload ; 
    }}

        
        
    
